play/src/front/Chat/Components/Room/RoomTimeline.svelte

```diff
@@ -63,27 +82,51 @@
     function handleScroll() {
         clearTimeout(scrollTimer);
         scrollTimer = setTimeout(() => {
-            if (messageListRef.scrollTop === 0) {
-                room.loadMorePreviousMessages().catch((error) => console.error(error));
+            const messages = Array.from(messageListRef?.querySelectorAll("li") || []);
+            const messagesBeforeViewportTop = messages.find((msg) => msg.getBoundingClientRect().top >= 0);
+            if (
+                messagesBeforeViewportTop &&
+                messages.indexOf(messagesBeforeViewportTop) < 10 &&
+                messages.indexOf(messagesBeforeViewportTop) !== -1
+            ) {
+                if (messageListRef.scrollTop === 0) isLoadingMessages = true;
+                room.loadMorePreviousMessages()
+                    .catch((error) => console.error(error))
+                    .finally(() => {
+                        if (isLoadingMessages) isLoadingMessages = false;
```

Comment:
Open question: should'nt we trigger "handleScroll()" at this point again? In case the number of loaded message is not enough to reach the top of the screen? (that could happen if there are a lot of reactions or message edition for instance, because those are not adding a lot of height to the viewport) Maybe it is handled somewhere else?